### PR TITLE
[receiver/chrony] generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add generated tests for chrony receiver; move initialization of the chrony client to the start function

**Link to tracking Issue:**
Relates to #27849

### CODE REVIEW

File: receiver/chronyreceiver/factory.go [hunk 1]
```diff
@@ -34,13 +34,15 @@ func newMetricsReceiver(
...
-	}
+	s := newScraper(ctx, cfg, set)
 	scraper, err := scraperhelper.NewScraper(
 		metadata.Type.String(),
-		newScraper(ctx, chronyc, cfg, set).scrape,
+		s.scrape,
+		scraperhelper.WithStart(func(ctx context.Context, host component.Host) error {
+			chronyc, err := chrony.New(cfg.Endpoint, cfg.Timeout)
+			s.client = chronyc
+			return err
```
Comment (dmitryax) [inline receiver/chronyreceiver/factory.go:44]: Can we move it to s.start?

**Review by dmitryax — APPROVED** [review]

File: receiver/chronyreceiver/metadata.yaml [hunk 2]
```diff
@@ -74,3 +74,7 @@ metrics:
       value_type: double
     attributes:
     - leap.status
+
+tests:
+  config:
+  skip_lifecycle: true
```
Comment (dmitryax) [inline receiver/chronyreceiver/metadata.yaml:80]: Why is this `true`?

File: receiver/chronyreceiver/factory.go (hunk 1 above)
Comment (dmitryax) [reply receiver/chronyreceiver/factory.go:44]: I was thinking if we can add `cfg` to `chronyScraper` struct and add a `start` method. But I'm good with this inline function as well

File: receiver/chronyreceiver/metadata.yaml (hunk 2 above)
Comment (atoulme) [reply receiver/chronyreceiver/metadata.yaml:80]: If you try to start the receiver, it fails looking for the chrony socket file.

Comment (atoulme) [reply receiver/chronyreceiver/metadata.yaml:80]: The generated tests test two scenarios:
* lifecycle, meaning we start and stop the receiver with a configuration, and make sure it is able to run. This is skipped.
* shutdown: we create the receiver and call .Shutdown on it. We check the receiver doesn't panic on this unexpected call.

Comment (dmitryax) [reply receiver/chronyreceiver/metadata.yaml:80]: I know. so the shutdown part is failing without functional changes?

Comment (atoulme) [reply receiver/chronyreceiver/metadata.yaml:80]: Indeed, the current behavior of the chrony receiver is to create a chrony client before creating the receiver. This client immediately looks up a file on disk and fails. Moving the chrony client creation to start helps.

Comment (dmitryax) [reply receiver/chronyreceiver/metadata.yaml:80]: Thanks for clarifying 👍

**Pull request merged**